Update gradle to 9.2.0

cppwfs opened the pull request:
Attempted to resolve TODO on line 1099, however some of the API is still not recognized.  Added a 9.2+ to the TODO

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

cppwfs committed:
Update gradle to 9.2.0

Attempted to resolve TODO on line 1099, however some of the API is
still not recognized.  Added a 9.2+ to the TODO

artembilan reviewed (changes requested):
Thanks, but I would like to know what is that new problem. The SF has gotten the fix though: https://github.com/spring-projects/spring-framework/blob/main/framework-api/framework-api.gradle

cppwfs committed:
Remove Gradle workaround from schema processing

The TODO comment indicated this workaround was needed until Gradle
9.1.1 or 9.2. Since Gradle has been updated to 9.2.0 in the
workaround is no longer necessary.

This change simplifies the schema loading code by using the more
concise `?.withInputStream` pattern instead of the verbose null-check
and conditional logic that was required by earlier Gradle versions.
